#!/usr/bin/env python

PACKAGE='platform_motion'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi
from sys import exit

from dynamic_reconfigure.parameter_generator import ParameterGenerator, double_t
from driver_base.msg import SensorLevels

gen = ParameterGenerator()

gen.add( "port_steering_offset", double_t,
        SensorLevels.RECONFIGURE_RUNNING, "Port steering offset",
        0, -pi/10, pi/10)
gen.add( "starboard_steering_offset", double_t,
        SensorLevels.RECONFIGURE_RUNNING, "Starboard steering offset",
        0, -pi/10, pi/10)
gen.add( "stern_steering_offset", double_t,
        SensorLevels.RECONFIGURE_RUNNING, "Stern steering offset",
        0, -pi/10, pi/10)
gen.add( "carousel_offset", double_t,
        SensorLevels.RECONFIGURE_RUNNING, "carousle position offset",
        0, -pi, pi)
gen.add( "wheel_diameter",  double_t,
        SensorLevels.RECONFIGURE_RUNNING, "Vehicle wheel diameter",
        0.33, 0.2, 0.4)

exit(gen.generate(PACKAGE, "motion_node", "PlatformParameters"))
